testdata/map.go

```diff
@@ -109,6 +112,48 @@ func main() {
 	squares = make(map[int]int, 20)
 	testBigMap(squares, 40)
 	println("tested growing of a map")
+
+	floatcmplx()
+}
+
+func floatcmplx() {
+
+	var zero float64
+	var negz float64 = math.Copysign(zero, -1)
```

Comment:
Yeah, I wasn't a huge fan of the hard-coded hash values either.  I'll rewrite this with option number (2).

Reply:
Hmm... just realized we don't want to actually modify the key value that's being passed in, which resetting the value via the pointer I believe is doing.

Reply:
Oops, you're right. This works around that:

```go
case reflect.Float32:
    f := *(*uint32)(ptr)
    if f == 0 {
        f = 0x80000000 // -0.0
    }
    return hashmapHash(unsafe.Pointer(&f), 4)
```

Reply:
Sidenote: I changed +0.0 to -0.0 in the example because comparing against 0 is slightly more efficient in many instruction sets. It's a micro-optimization though, and probably doesn't make much difference.

Reply:
I can only imagine that the difference in comparing against zero vs not is totally outweighed by the actual hash function calculation.

Reply:
Hmm... my approach of having two hash function calls probably has additional code side overhead than just resetting the value.  I'll update my PR.